src/linear_algebra/general_linear_group.lean

```diff
@@ -84,6 +89,21 @@ begin
   exact inv_of_eq_nonsing_inv (↑A : matrix n n R),
 end
 
+/-- An element of the matrix general linear group on `(n) [fintype n]` can be considered as an
+element of the endomorphism general linear group on `n → R`. -/
+def to_linear : general_linear_group n R ≃* linear_map.general_linear_group R (n → R) :=
+units.map_equiv matrix.to_lin_alg_equiv'.to_ring_equiv.to_mul_equiv
+
+-- For some reason this doesn't trigger as a simp lemma.  Decidability issue apparently?
+-- TODO fix this and delete the special case stated for `n` rather than `fin m` below.
+lemma coe_to_linear (M : general_linear_group n R) :
+  (to_linear M : (n → R) →ₗ[R] (n → R)) = matrix.mul_vec_lin M :=
+rfl
+
```

Comment:
```suggestion

@[simp] lemma to_linear_apply (M : @general_linear_group n R ‹_› _ _) (v : n → R) :
  (@general_linear_group.to_linear n ‹_› ‹_› R _ M) v = matrix.mul_vec_lin M v :=
rfl
```
where the same remark about `‹_›` and decidability arguments applies.

Reply:
@eric-wieser, thanks for tracking this issue down.  I found that it works without using @ and _ on the first argument, i.e. to take just `(M : general_linear_group n R)` rather than `(M : @general_linear_group n R ‹_› _ _)`.  OK with you?

Reply:
Yes, I found it worked too, but it produces a slightly more worrying output for `#check` that still has the lambda terms in the type of the `M` argument. Leaving it as it was maybe leaves more of a breadcrumb for someone who runs into this again, but perhaps the place for that is a MWE in a new github issue. Certainly it feels nicer to avoid underscores.

Perhaps as a compromise add to the comment the fact that "these lambda terms also appear in the type of `M`, but as simp doesn't get confused by then we do not worry"?

Reply:
Added